### PR TITLE
chore: use sourcesJar and javadocJar tasks provided by Gradle 6

### PR DESCRIPTION
**Pull request by mamadzz**
Issue: https://github.com/ekino/spring-boot-starter-errors/issues/17

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -112,29 +104,12 @@ tasks {
...
-        url = URL("https://docs.spring.io/spring-framework/docs/5.1.9.RELEASE/javadoc-api/")
-        packageListUrl = URL(url, "package-list")
-      }
-      externalDocumentationLink {
-        url = URL("https://docs.spring.io/spring-boot/docs/2.1.x/api/")
-        packageListUrl = URL(url, "package-list")
-      }
-    }
-  }
-
```
Comment (clemstoquart) [inline build.gradle.kts]: Hi, I don't understand why you deleted that. We don't want to lose the project documentation.

Comment (mamadzz) [reply build.gradle.kts]: Hi, I had inadvertently deleted it (during my tests).
Thanks for your return.

File: build.gradle.kts [hunk 2]
```diff
@@ -88,15 +88,9 @@ val build by tasks.named("build") {
...
-  archiveClassifier.set("sources")
-  from(sourceSets.main.get().allJava)
-}
-
-val javadocJar by tasks.registering(Jar::class) {
-  dependsOn("dokka")
-  archiveClassifier.set("javadoc")
-  from(buildDir.resolve("dokka"))
+java {
+  withJavadocJar()
```
Comment (clemstoquart) [inline build.gradle.kts]: We need a way to keep the `dependsOn("dokka")`.

Comment (mamadzz) [reply build.gradle.kts]: Ok

Comment (clemstoquart) [reply build.gradle.kts]: I'm afraid it's not enough. The generated javadoc jar is empty.

Comment (mamadzz) [reply build.gradle.kts]: Hello, in fact. For now it seems that the generation of doc with gradle 6 withJavadocJar does not work well with dokka. I think at first we will keep the generation task that you have implemented.

Comment (clemstoquart) [reply build.gradle.kts]: Ok. Thanks.

**Pull request merged**